zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/repo/HDFSNotebookRepo.java

```diff
@@ -0,0 +1,503 @@
+      UserGroupInformation.setConfiguration(conf);
+      try {
+        UserGroupInformation.loginUserFromKeytab(
+            this.conf.getKerberosPrincipal(),
+            this.conf.getKerberoskeyTab()
+        );
+
+        if (UserGroupInformation.isLoginKeytabBased()) {
+          UserGroupInformation.getLoginUser().reloginFromKeytab();
+          return true;
+        } else if (UserGroupInformation.isLoginTicketBased()) {
+          UserGroupInformation.getLoginUser().reloginFromTicketCache();
+          return true;
+        }
+
+      } catch (IOException e) {
+        LOG.error("Failed to get either keytab location or principal name in the " +
+            "interpreter", e);
+      }
+      return true;
+    } catch (Exception e) {
+      LOG.error("Unable to run kinit for zeppelin", e);
+    }
+    return false;
+  }
+
+  private void setNotebookDirectory(String notebookDirPath) throws IOException {
+    stringPath = notebookDirPath;
+    setNotebookDirPath(getPath(stringPath));
+    if (!fileSystem.exists(getRootDir())) {
+      LOG.info("Notebook dir doesn't exist, create on is {}.", stringPath);
+      createFolder(this.notebookDirPath);
+    }
+  }
+
+  protected Path getRootDir() throws IOException {
+    if (!fileSystem.exists(fileSystem.getHomeDirectory())) {
+      throw new IOException("Root path does not exists");
+    }
+    if (!isDirectory(fileSystem.getHomeDirectory())) {
+      throw new IOException("Root path is not a directory");
+    }
+    return getNotebookDirPath();
+  }
+
+  private Path getPath(String path) {
+    String absolutePath;
+    if (path == null || path.trim().length() == 0) {
+      absolutePath = fileSystem.getHomeDirectory().toString();
+    } else if (path.startsWith("/")) {
+      absolutePath = fileSystem.getHomeDirectory().toString() + path;
+    } else {
+      absolutePath = fileSystem.getHomeDirectory().toString() + "/" + path;
+    }
+    return new Path(absolutePath);
+  }
+
+  private boolean isDirectory(Path path) throws IOException {
+    if (fileSystem == null) {
+      return false;
+    }
+    FileStatus fileStatus = null;
+    try {
+      fileStatus = fileSystem.getFileStatus(path);
+    } catch (FileNotFoundException e) {
+      LOG.info("File/Directory not found, nothing to worry:" + path);
+      return false;
+    }
+    if (fileStatus.isDirectory()) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  private void createFolder(Path path) throws IOException {
+    fileSystem.mkdirs(path);
+  }
+
+  @Override
+  public List<NoteInfo> list(AuthenticationInfo subject) throws IOException {
+    FileStatus[] children = fileSystem.listStatus(getRootDir());
+
+    List<NoteInfo> infos = new LinkedList<>();
+    for (FileStatus f : children) {
+      String fileName = f.getPath().getName();
+      if (fileName.startsWith(".")
+          || fileName.startsWith("#")
+          || fileName.startsWith("~")) {
```

Comment:
shouldn't this be `_` and `.` only
https://github.com/apache/hadoop/blob/trunk/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapred/FileInputFormat.java#L92